src/popup/components/Item/index.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react'
+import {KeyboardEvent, ChangeEvent} from 'react';
+import * as classnames from 'classnames'
+
+export interface ItemProps {
+    id?: number | string
+    selected: boolean
+    favIconUrl: string
+    title: string
+    url: string
+    itemType: ItemType
+}
+  
+export interface ItemState {}
+
+export default class Item extends React.Component<ItemProps, ItemState> {
+
+    element: HTMLElement
+
+    static buildFromTab(item: chrome.tabs.Tab, isSelected: boolean): JSX.Element {
+        return this.build(item, ItemType.Tab, isSelected);
+    }
+
+    static buildFromBookmark(item: chrome.bookmarks.BookmarkTreeNode, isSelected: boolean) {
+        return this.build(item, ItemType.Bookmark, isSelected);
+    }
+
+    private static build(item: chrome.bookmarks.BookmarkTreeNode | chrome.tabs.Tab, itemType: ItemType, isSelected: boolean): JSX.Element {
```

Comment:
TBH, I hate knowing the type by sending it as a parameter (`itemType`), but I had trouble using `instanceof`. Checked several websites and tutorials and couldn't make it work. Also used `typeof` and no luck. Suggestions?